### ISSUE TITLE
Clean up improved renderer integration

### ISSUE THREAD

**Issue by graynorton** [PolymerLabs/uni-virtualizer#46]
EDIT May 12: Changing the scope of this issue to reflect a change in plans. Got the cleanup done in Sprint 27, but holding off on landing until some additional work is done (most notably finalizing the browser support and polyfill story).

## Description

The purpose of this task to clean up ~~and land~~ the refactoring we did in #41. The refactoring itself is expected to make the code smaller and simpler, laying a better foundation for future work. Our immediate goal in landing the refactoring (the work in this task) is to unblock internal and external adoption of the library as we move toward a 1.0 release.

The refactoring is currently on the `better-grid` branch, mixed with some new layout features and miscellaneous bug fixes.

In this task, we'll:

- Clean up the refactoring and remove older code made redundant
- ~~Separate the new layout features into a different branch~~
- ~~Prepare one or more PRs to land the bug fixes and refactoring~~

## Acceptance criteria

~~By the end of the sprint, we want to have one or more review-ready PRs to land the refactoring and bug fixes.~~

By the end of the sprint, we'll have the newly factored implementation cleaned up and the old implementation removed.
